CI failure in `Validate TeamCity Configuration` workflow

[For example](https://github.com/hashicorp/terraform-provider-aws/actions/runs/25164753163/job/73767889993):

```
Error:  Failed to execute goal org.jetbrains.teamcity:teamcity-configs-maven-plugin:2026.1-eap17:generate (default-cli) on project TerraformProviderAWS: 
Error:  The plugin org.jetbrains.teamcity:teamcity-configs-maven-plugin:2026.1-eap17 has unmet prerequisites: 
Error:  	Required Java version 21 is not met by current version: 17.0.19
Error:  -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginIncompatibleException
make: *** [GNUmakefile:7: validate] Error 1
Error: Process completed with exit code 2.
```

https://www.jetbrains.com/help/teamcity/upgrade-notes.html#2025.11:

> The migration date for Java 21 is now finalized: starting with version 2026.1, TeamCity servers and agents will not be able to start under Java versions older than 21.

--- a/.github/workflows/gen-teamcity.yml
+++ b/.github/workflows/gen-teamcity.yml
@@ -20,7 +20,7 @@ jobs:
       - uses: actions/setup-java@be666c2fcd27ec809703dec50e508c2fdc7f6654 # v5.2.0
         with:
           distribution: adopt
-          java-version: 17
+          java-version: 21
           cache: maven
       - name: Build TeamCity Configuration
         run: |

--- a/.teamcity/.java-version
+++ b/.teamcity/.java-version
@@ -1,1 +1,1 @@
-corretto64-17.0.7
+corretto64-21.0.11

